src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -156,76 +138,97 @@ def _get_scope(self) -> str:
             self._workspace_name,
         )
 
-    def _start_run(self, run_name: Optional[str]) -> bool:
+    def _start_run(self) -> None:
         """
-        Make a request to start the mlflow run. If the run will not start, it will be
-
-        marked as broken and the logging will be switched off.
-        :param run_name: The display name for the run.
-        :type run_name: Optional[str]
-        :returns: True if the run has started and False otherwise.
+        Start the run, or, if it is not applicable (for example, if tracking is not enabled), mark it as started.
         """
-        url = f"https://{self._url_base}/mlflow/v2.0" f"{self._get_scope()}/api/2.0/mlflow/runs/create"
-        body = {
-            "experiment_id": "0",
-            "user_id": "promptflow-evals",
-            "start_time": int(time.time() * 1000),
-            "tags": [{"key": "mlflow.user", "value": "promptflow-evals"}],
-        }
-        if run_name:
-            body["run_name"] = run_name
-        response = self.request_with_retry(
-            url=url,
-            method='POST',
-            json_dict=body
-        )
-        if response.status_code != 200:
-            self.info = RunInfo.generate(run_name)
-            LOGGER.warning(f"The run failed to start: {response.status_code}: {response.text}."
+        if self._status != RunStatus.NOT_STARTED:
+            raise ValueError("The run has already started. Please end this run and to start another one.")

```

Comment:
If the incorrect state is due to an inappropriate value being passed to a function, then ValueError makes sense. Otherwise, I think RuntimeError is probably a better fit.

And, can we reuse `_check_state_and_log` here as well?

Reply:
Fixed.